wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginManager.java

```
@@ -251,6 +290,30 @@ protected <T, E extends Exception> T executeWithSubscribedPlugins(
     return pluginChainFunc.call(pluginPipeline, jdbcMethodFunc);
   }
 
+
+  protected <T, E extends Exception> T executeWithTelemetry(
+      final @NonNull JdbcCallable<T, E> execution,
+      final @NonNull String pluginName) throws E {
+    TelemetryContext context = null;
+    try {
+      try {
+        final String contextName = pluginName;
+        context = telemetryFactory.openTelemetryContext(contextName);
```

Can we hide exception handling inside openTelemetryContext() ?